### PR TITLE
Feature/axelar query migration

### PR DESCRIPTION
**Pull request by SGiaccobasso**
# Description

Migrated the `getNativeGasBaseFee ` function from SDK to the API package.

## ⛑️ **What was done:**

Created the function, corrected typescript errors, created auxiliary files and added dependencies.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update
- [ ] Ready to merge

### 🧪 **How to test:**

There's a test for this function in `axelarjs/packages/api/src/axelar-query/client.spec.ts`, you can run it from `axelarjs/packages/api` with the command `pnpm run test`.

### 🗒️ **Notes:**

- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by npty — COMMENTED** [review]
I noticed that there're lots of types migrated from the SDK v1. Can you add only functions that the function `getNativeBaseFee` relies on for now?  However, I think we can keep the `validateChain` utils. We may use it to improve our validation logic.

File: packages/api/src/axelar-query/isomorphic.ts [hunk 1]
```diff
@@ -319,4 +345,144 @@ export class AxelarQueryAPIClient extends RestService {
...
+        await AxelarQueryClient.initOrGetAxelarQueryClient({
+          environment: this.env,
+          axelarRpcUrl: this.axelarRpcUrl,
+        });
+    }
+
+    return this.axelarQueryClient.nexus
+      .Chains({ status: ChainStatus.CHAIN_STATUS_ACTIVATED })
+      .then((resp) => resp.chains);
+  }
```
Comment (npty) [inline packages/api/src/axelar-query/isomorphic.ts:365]: i think we can remove the `AxelarQueryClient` and use [the existing cosmos package](https://github.com/axelarnetwork/axelarjs/tree/main/packages/cosmos) instead

File: packages/api/src/axelar-query/isomorphic.ts [hunk 2]
```diff
@@ -319,4 +345,144 @@ export class AxelarQueryAPIClient extends RestService {
...
+  public async getNativeGasBaseFee(
+    sourceChainId: EvmChain,
+    destinationChainId: EvmChain,
+    sourceTokenSymbol?: GasToken,
+    symbol?: string,
+    destinationContractAddress?: string,
+    sourceContractAddress?: string,
+    amount?: number,
+    amountInUnits?: bigint | string
+  ): Promise<BaseFeeResponse> {
```
Comment (npty) [inline packages/api/src/axelar-query/isomorphic.ts:412]: can you group this optional params into options object?

File: packages/api/src/axelar-query/isomorphic.ts [hunk 3]
```diff
@@ -319,4 +345,144 @@ export class AxelarQueryAPIClient extends RestService {
...
+    if (symbol) params.symbol = symbol;
+    if (destinationContractAddress)
+      params.destinationContractAddress = destinationContractAddress;
+    if (sourceContractAddress)
+      params.sourceContractAddress = sourceContractAddress;
+    if (amount) {
+      params.amount = amount;
+    } else if (amountInUnits) {
+      params.amountInUnits = amountInUnits;
+    }
```
Comment (npty) [inline packages/api/src/axelar-query/isomorphic.ts:441]: can you clean this up? does it work if we don't check if the variable is defined before assigning it to params?

File: packages/api/src/axelar-query/isomorphic.ts [hunk 4]
```diff
@@ -319,4 +345,144 @@ export class AxelarQueryAPIClient extends RestService {
...
+  }
+
+  /**
+   * Gets the base fee in native token wei for a given source and destination chain combination
+   * @param sourceChainName
+   * @param destinationChainName
+   * @param sourceTokenSymbol (optional)
+   * @returns base fee in native token in wei, translated into the native gas token of choice
+   */
+  public async getNativeGasBaseFee(
```
Comment (npty) [inline packages/api/src/axelar-query/isomorphic.ts:403]: I think we can improve this from what it is in sdk v1. will provide few improvement suggestions below.

File: packages/api/src/axelar-query/assets/index.ts [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    });
+
+    const result = (await response.json()) as { [key: string]: AssetConfig };
+
+    return result;
+  } catch (error) {
+    console.error({ error });
+    throw error;
+  }
+}
```
Comment (npty) [inline packages/api/src/axelar-query/assets/index.ts:51]: I think we can use [the existing AxelarConfigClient](https://github.com/axelarnetwork/axelarjs/blob/main/packages/api/src/axelar-config/isomorphic.ts) for this file instead.

File: packages/api/src/axelar-query/isomorphic.ts (hunk 4 above)
Comment (SGiaccobasso) [reply packages/api/src/axelar-query/isomorphic.ts:411]: let me know if it's better now, and if it still needs changes

File: packages/api/src/axelar-query/chains/index.ts [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+      config.environment as Environment
+    )) as ChainInfo
+  ) as ChainInfo[];
+
+  /*push assets to supported chains*/
+  rawChains.forEach((chainInfo) => {
+    const filteredAssetList: AssetConfig[] = allAssets.filter(
+      ({ chain_aliases }) =>
+        Object.keys(chain_aliases).indexOf(chainInfo.chainName.toLowerCase()) >
+        -1
```
Comment (npty) [inline packages/api/src/axelar-query/chains/index.ts:35]: Perhaps this? 
```ts
Object.keys(chain_aliases).includes(chainInfo.chainName.toLowerCase())
```

File: packages/api/src/axelar-query/chains/index.ts [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+      ) as AssetInfo;
+      assetToPush.common_key = asset.common_key[_environment] as string;
+      assetToPush.native_chain = asset.native_chain;
+      assetToPush.decimals = asset.decimals;
+      assetToPush.fullySupported = asset.fully_supported;
+      assetsList.push(assetToPush);
+    });
+
+    chainInfo.assets = assetsList;
+  });
```
Comment (npty) [inline packages/api/src/axelar-query/chains/index.ts:52]: just learn that we have `clone-deep` dependency. can we try using object destructuring instead? so we might be able to remove this dependency later.

example

```ts
 const {
    chain_aliases,
    common_key,
    native_chain,
    decimals,
    fully_supported: fullySupported
  } = asset;

  const assetToPush = {
    ...chain_aliases[chainInfo.chainName.toLowerCase()],
    common_key: common_key[_environment],
    native_chain,
    decimals,
    fullySupported
  };

  assetsList.push(assetToPush);
  ```

File: packages/api/src/axelar-query/chains/supported-chains-list.ts [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+    SOMMELIER: "sommelier",
+    STARGAZE: "stargaze",
+    STRIDE: "stride",
+    TERITORI: "teritori",
+    TERRACLASSIC: "terra",
+    TERRA: "terra-2",
+    UMEE: "umee",
+    XPLA: "xpla",
+  },
+};
```
Comment (npty) [inline packages/api/src/axelar-query/chains/supported-chains-list.ts:111]: i think we don't want to maintain this file, so we can remove it.

File: packages/api/src/axelar-query/types.ts [hunk 9]
```diff
@@ -20,3 +21,166 @@ export type EstimateGasFeeResponse = {
...
+  gas_price: string;
+  decimals: number;
+  name: string;
+  l1_gas_oracle_address?: string;
+  l1_gas_price_in_units?: TokenUnit;
+  symbol: string;
+  token_price: {
+    usd: number;
+  };
+};
```
Comment (npty) [inline packages/api/src/axelar-query/types.ts:108]: i think we don't need this, please check others interface and types as well.

File: packages/api/src/axelar-query/isomorphic.ts [hunk 10]
```diff
@@ -319,4 +347,142 @@ export class AxelarQueryAPIClient extends RestService {
...
+   *   - symbol
+   *   - destinationContractAddress
+   *   - sourceContractAddress
+   *   - amount
+   *   - amountInUnits
+   * @returns A `BaseFeeResponse` object containing the base fee, express fee, source token information, execute gas multiplier, destination token information, L2 type, Ethereum token information, and success status.
+   */
+  public async getNativeGasBaseFee(
+    sourceChainId: EvmChain,
+    destinationChainId: EvmChain,
```
Comment (npty) [inline packages/api/src/axelar-query/isomorphic.ts:413]: update type from `EvmChain` to `string` for both params

File: packages/api/package.json [hunk 11]
```diff
@@ -105,8 +108,14 @@
...
     "@axelarjs/cosmos": "workspace:*",
     "@axelarjs/utils": "workspace:*",
+    "@axelarjs/proto": "workspace:*",
+    "clone-deep": "^4.0.1",
+    "cross-fetch": "^4.0.0",
+    "ethers": "^6.13.2",
     "isomorphic-unfetch": "^4.0.2",
     "rambda": "^9.1.1",
+    "string-similarity-js": "^2.1.4",
+    "standard-http-error": "^2.0.1",
```
Comment (npty) [inline packages/api/package.json:118]: i think only `string-similarity-js` is necessary here. could you try removing others?

File: packages/api/src/axelar-query/types.ts [hunk 12]
```diff
@@ -20,3 +28,83 @@ export type EstimateGasFeeResponse = {
...
+export interface AxelarGMPResponse {
+  result: AxelarGMPResult;
+}
+export interface AxelarQueryService {
+  readonly evm: EvmQS;
+  readonly axelarnet: AxelarnetQS;
+  readonly nexus: NexusQS;
+  readonly tss: TSSQS;
+}
+export type AxelarQueryClientType = QueryClient & AxelarQueryService;
```
Comment (npty) [inline packages/api/src/axelar-query/types.ts:110]: can you please remove interfaces/types that we haven't used in this PR?

Comment (SGiaccobasso) [reply packages/api/src/axelar-query/types.ts:110]: all the types left here are being used

Comment (npty) [reply packages/api/src/axelar-query/types.ts:110]: thanks for the check. there're some fields duplicated with some types [here](https://github.com/axelarnetwork/axelarjs/blob/main/packages/api/src/gmp/types.ts). can you check if there's anyway we can combine or reuse them?

File: packages/api/src/axelar-query/isomorphic.ts [hunk 13]
```diff
@@ -319,4 +341,138 @@ export class AxelarQueryAPIClient extends RestService {
...
   }
+
+  /**
+   * Get a list of active chains.
+   * @returns an array of active chains
+   */
+  public async getActiveChains(): Promise<string[]> {
+    const axelarQueryRpcClient = (await createAxelarRPCQueryClient({
+      environment: this.env,
+    })) as AxelarQueryClientType;
```
Comment (npty) [inline packages/api/src/axelar-query/isomorphic.ts:352]: can we reuse [this type](https://github.com/axelarnetwork/axelarjs/blob/76f4a83a5c808753e7e0567c9862c5ca68b3a656/packages/cosmos/src/rpc/types.ts#L100)?

File: packages/api/src/axelar-query/types.ts [hunk 14]
```diff
@@ -20,3 +28,83 @@ export type EstimateGasFeeResponse = {
   apiResponse: string;
   isExpressSupported: boolean;
 };
+
+export type Environment = "devnet" | "testnet" | "mainnet";
```
Comment (npty) [inline packages/api/src/axelar-query/types.ts:32]: use [this type](https://github.com/axelarnetwork/axelarjs/blob/76f4a83a5c808753e7e0567c9862c5ca68b3a656/packages/cosmos/src/types.ts#L3) instead.

**Review by npty — APPROVED** [review]

**Pull request merged**